pycqed/instrument_drivers/meta_instrument/qubit_objects/qubit_object.py

```diff
@@ -386,32 +386,37 @@ def find_frequency(self, method='spectroscopy', pulsed=False,
     def find_resonator_frequency(self, **kw):
         raise NotImplementedError()
 
-    def find_pulse_amplitude(self, amps,
+    def find_pulse_amplitude(self, amps=np.linspace(-.5, .5, 31),
                              N_steps=[3, 7, 13, 17], max_n=18,
                              close_fig=True, verbose=False,
                              MC=None, update=True, take_fit_I=False):
         '''
-        Finds the pulse-amplitude using a rabi experiment.
+        Finds the pulse-amplitude using a Rabi experiment.
+        Fine tunes by doing a Rabi around the optimum with an odd
+        multiple of pulses.
 
-        If amps is an array it starts by fitting a cos to a Rabi experiment
-        to get an initial guess for the amplitude.
-        If amps is a float it uses that as the initial amplitude and starts
-        doing rabi flipping experiments around that optimum directly.
+        Args:
+            amps: (array or float) amplitudes of the first Rabi if an array,
+                if a float is specified it will be treated as an estimate
+                for the amplitude to be found.
+            N_steps: (list of int) number of pulses used in the fine tuning
+            max_n: (int) break of if N> max_n
         '''
         if MC is None:
             MC = self.MC
         if np.size(amps) != 1:
             self.measure_rabi(amps, n=1, MC=MC, analyze=False)
             a = ma.Rabi_Analysis(close_fig=close_fig)
+            # Decide which quadrature to take by comparing the contrast
             if take_fit_I:
-                ampl = abs(a.fit_res[0].params['period'].value)/2
-                print("taking I")
+                ampl = abs(a.fit_res[0].params['period'].value)/2.
+            elif (np.abs(max(a.fit_res[0].data) -
```

Comment:
You are essentially taking the amplitude of the data here but taking it from the fit result. I would suggest changing this to directly taking it from the data (without the a.fit_res[0]), as this will only create confusion.  

I did not change this as I cannot test the qubit object but I would suggest we change this.